### PR TITLE
Add missing cli flag as configuration options

### LINKED ISSUE #968: Allow `compile` cli options to be specified in `cadl-project.yaml`
| Cli options     | Config name    | Added |
| --------------- | -------------- | --------------- |
| `--emit`        | `emitters`     | ✅              |
| `--option`      | via `emitters` | ✅              |
| `--import`      | `imports`      |
| `--output-path` | `output-path`  |
| `--trace`       | `trace`        |
| `--watch`       | ❌             |
| `--pretty`      | ❌             |

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #968

### CODE REVIEW

File: packages/compiler/core/cli.ts [hunk 1]
```diff
@@ -365,16 +361,21 @@ async function getCompilerOptions(
...
+  const outputPath = resolvePath(process.cwd(), pathArg);
+  await mkdirp(outputPath);
+
   const cliOptions = resolveOptions(args);
+
   return {
     outputDir: outputPath,
     nostdlib: args["nostdlib"],
-    additionalImports: args["import"],
+    additionalImports: args["import"] ?? config["imports"],
```
Comment (nguerrera) [inline packages/compiler/core/cli.ts:373]: It feels like these should be able to combine between CLI and config file.

Comment (nguerrera) [reply packages/compiler/core/cli.ts:373]: But then should there be a way to clear the config. I'm not sure here.

File: packages/compiler/config/config-schema.ts [hunk 2]
```diff
@@ -9,6 +9,28 @@ export const CadlConfigJsonSchema: JSONSchemaType<CadlRawConfig> = {
...
+    },
+    trace: {
+      oneOf: [
+        { type: "string" },
+        {
+          type: "array",
+          items: { type: "string" },
+        },
+      ],
+    } as any,
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/compiler/config/config-schema.ts:28]: this is an unfortunate issue with the `ajv` type that require an optional property to also be marked as `nullable: true` but then the actual engine then complains that we are using `nullable` without `type` (as its oneOf here)

File: packages/compiler/config/config-loader.ts [hunk 3]
```diff
@@ -111,7 +111,18 @@ async function loadConfigFile(
...
+  return cleanUndefined({
+    filename,
+    diagnostics,
+    outputDir: data["output-dir"],
+    warnAsError: data["warn-as-error"],
+    imports: data.imports,
+    extends: data.extends,
+    trace: typeof data.trace === "string" ? [data.trace] : data.trace,
+    emitters: data.emitters!,
+  });
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/compiler/config/config-loader.ts:123]: yeah

File: packages/compiler/core/cli.ts (hunk 1 above)
Comment (timotheeguerin) [reply packages/compiler/core/cli.ts:373]: yeah everything in the config is override for now. Same as tsconfig, they did have an issue to add a way to append(when doing extends in particular) and it was rejected as it was adding too much complexcity(A little duplication sometimes isn't bad)

File: packages/compiler/config/config-schema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nguerrera) [reply packages/compiler/config/config-schema.ts:28]: Could you add a comment on this?

[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/compiler/config/config-schema.ts]: yep added a comment with a link to ajv issue

**Review by nguerrera — APPROVED** [review]

**Pull request merged**